### ISSUE TITLE
Hi, @oblomov-dev

### ISSUE THREAD

**Issue by Hhpy410** [abap2UI5/abap2UI5#1127]
Hi, @oblomov-dev
This is a great project.
I have an additional question.
How can we download the Properties, Aggregations, Events, Methods of any components?
If this is possible We would like to make a workbench like Webdynpro workbench.

![1](https://github.com/abap2UI5/abap2UI5/assets/38339002/b29473f1-52ab-44b9-9fce-21d3a390e6c1)

**Comment by mbtools** [comment]
No idea but I do know how to ask ChatGPT 😉 

![image](https://github.com/abap2UI5/abap2UI5/assets/59966492/41656c71-1c32-453b-816b-54b5db2e4501)

**Comment by choper725** [comment]
what is the idea to build with that. ? as this will happen in the client side.. what should be done with this ?

**Comment by oblomov-dev** [comment]
@mbtools mhh never thought about doing it this way, but interesting idea, thank you, will take a look to it.

@choper725 some stuff in abap2UI5 is just a copy of the UI5 API. For example the class z2ui5_cl_xml_view, creating this ABAP artifact automatically could save us a lot of work. Or see the idea of @Hhpy410

yes having this at the client side is a limitation, best would be having a REST API giving you the possibility to call it with the language of your choice. But as a start, better than nothing...

**Comment by choper725** [comment]
like calling an api to get all properties and send the ones you need to abap2ui5 xml builder ?!

sounds intresting….

**Comment by mbtools** [comment]
You do it as a node action in a repo, generate the interface.abap file(s), and clone them to your project. It won't work with the ui5 runtime since it's minimized (i.e. cryptic parameter names) but should be feasible with the full ui5 repo. 

However, there's another, even better way: The UI5 docs are generated using [JSdoc](https://github.com/jsdoc/jsdoc). Why not create a JSdoc template that outputs interface.abap as the content (red part)? 

![image](https://github.com/abap2UI5/abap2UI5/assets/59966492/e54e3d2f-a104-4cfc-abc7-2f07d65d1e85)

Start with the [minimal theme](https://github.com/Nijikokun/minami). Rewrite [this](https://github.com/nijikokun/minami/blob/master/tmpl/method.tmpl) to output the abap code (instead of html). 

You would get a usable website to search and browse the interfaces and could build tooling to clone and reuse the interfaces in your own project.

**Comment by Hhpy410** [comment]
Thanks @mbtools ,
I'm an abap developer so I didn't really understand it, 
but I think it would be helpful.
 I'll have to give it a try.

**Comment by oblomov-dev** [comment]
@mbtools yes this sounds like a great approach, some new stuff for me to learn, but i will take a look to it! thank you for extending my understanding how all these documentations are generated and work.

**Comment by oblomov-dev** [comment]
this issue got old.

With AI it is easily possible now — you can just point an LLM at the UI5 API docs (https://ui5.sap.com/#/api) and it will translate any control 1:1 into whatever you need. 

see for a translation to ABAP Doc: https://github.com/abap2UI5/abap2UI5/pull/2243

closing.
